### PR TITLE
MISRA fixes

### PR DESCRIPTION
**Pull request by bradleysmith23**
Description
-----------

MISRA violations identified through the use of Coverity and [here](https://github.com/FreeRTOS/coreJSON/issues/152)
Fixes to violations of MISRA rule 10.3 and rule 20.12 
MISRA rule 10.3 violations: assigning expression of essentially unsigned type to an object with essentially signed type. (int32_t value = 0U;) and assignment of int8_t to a char (char_ c = { 0 }; and char_ n = { 0 };) 

MISRA rule 20.12 violation: assert( isOpenBracket_( mode ) ); -> Here, macro parameter isOpenBracket_ (which is also macro) is itself subject for further macro replacement (#expr), supplied as expr parameter. 

Test Steps
-----------

Checklist:
----------

- [x] I have tested my changes. No regression in existing tests.
- [ ] I have modified and/or added unit-tests to cover the code changes in this Pull Request.

Related Issue
-----------

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by kstribrnAmzn — APPROVED** [review]

File: source/include/core_json.h [hunk 1]
```diff
@@ -39,6 +39,26 @@
...
+ * It is useful to set this to 1 while developing your application for
+ * debugging purposes.
+ */
+#define coreJSON_ASSERT_DEFINED    ( 0 )
+
+#if ( coreJSON_ASSERT_DEFINED == 1 )
+
+/**
+ * @brief Define assert_param() to sit in a loop if an assertion fails */
+    #define  assert_param( expr )    if( ( expr ) == ( bool ) 0 ) { for( ; ; ) {} }
```
[the remark this answers is not in the document]
Comment (paulbartell) [reply source/include/core_json.h:54]: What's the problem with using assert as-is?

**Review by paulbartell — CHANGES_REQUESTED** [review]
It's not clear why we need to change this. There must be a better way to satisfy the customer's MISRA checker.

**Review by kstribrnAmzn — APPROVED** [review]

**Review by moninom1 — APPROVED** [review]

**Comment by chinglee-iot** [comment]
MISRA C 2012 rule 20.12 deviation doesn't seem to be fixed when `#define configJSON_ASSERT( expr ) assert( exprt )`
Creating a new PR #157 to cover this PR.

**Review by paulbartell — APPROVED** [review]

**Pull request merged**